### DISCUSSION TITLE
Editor's "dirty" state

### DISCUSSION THREAD

**Discussion by evandor** [codex-team/editor.js#2466] [Q&A]
I wonder if I missed it - but it does not seem possible to query the editor's state, if it is "dirty" or not. By dirty I mean that a user has made changes without saving them yet. Ideally, if a user reverts a single change, the editor's state should switch back to "not dirty" again, as the content is the same again. 

This can be used to toggle a save button for example.

**Comment by neSpecc** [comment]
Interesting idea. You can implement it outside the editor: save the data, make a hash from it, and compare it with the previous hash.

**Reply by evandor** [reply]
That would work, but you'd have to do this for every single keystroke... an in-built feature could handle this much more efficiently I reckon.
